src/ai/backend/manager/models/vfolder.py

```diff
@@ -720,9 +716,11 @@ async def prepare_vfolder_mounts(
     if requested_vfolder_names:
         extra_vf_conds = vfolders.c.name.in_(
             requested_vfolder_names.values()
-        ) | vfolders.c.name.startswith(".")
+        ) | vfolders.c.name.startswith(".") & vfolders.c.status.not_in(DEAD_VFOLDER_STATUSES)
     else:
-        extra_vf_conds = vfolders.c.name.startswith(".")
+        extra_vf_conds = vfolders.c.name.startswith(".") & vfolders.c.status.not_in(
+            DEAD_VFOLDER_STATUSES
```

Comment:
I think that even when `requested_vfolder_names` are provided, VFolders should only be mounted if they are available. How about refactoring like below?

```py
    extra_vf_conds = vfolders.c.name.startswith(".") & vfolders.c.status.not_in(
        DEAD_VFOLDER_STATUSES
    )

    if requested_vfolder_names:
        extra_vf_conds = extra_vf_conds | (
            vfolders.c.name.in_(requested_vfolder_names.values())
            & vfolders.c.status.not_in(DEAD_VFOLDER_STATUSES)
        )
```

Reply:
Bulls eye! I'll apply it right away.